src/Cones/epipertraceentropytri.jl

```diff
@@ -0,0 +1,183 @@
+    use_dual_barrier::Bool
+    dim::Int
+    d::Int
+    rt2::T
+
+    point::Vector{T}
+    dual_point::Vector{T}
+    grad::Vector{T}
+    correction::Vector{T}
+    vec1::Vector{T}
+    vec2::Vector{T}
+    feas_updated::Bool
+    grad_updated::Bool
+    hess_updated::Bool
+    inv_hess_updated::Bool
+    hess_fact_updated::Bool
+    is_feas::Bool
+    hess::Symmetric{T, Matrix{T}}
+    inv_hess::Symmetric{T, Matrix{T}}
+    hess_fact_cache
+
+    W::Matrix{T}
+    Wi::Matrix{T}
+    fact_W
+    z::T
+    lwv::Matrix{T}
+    tau::Matrix{T}
+    sigma::T
+    w_vals_log::Vector{T}
+
+    function EpiPerTraceEntropyTri{T}(
+        dim::Int;
+        use_dual::Bool = false,
+        hess_fact_cache = hessian_cache(T),
+        ) where {T <: Real}
+        @assert dim >= 3
+        cone = new{T}()
+        cone.use_dual_barrier = use_dual
+        cone.dim = dim
+        cone.d = round(Int, sqrt(0.25 + 2 * (dim - 2)) - 0.5)
+        cone.rt2 = sqrt(T(2))
+        cone.hess_fact_cache = hess_fact_cache
+        return cone
+    end
+end
+
+use_heuristic_neighborhood(::EpiPerTraceEntropyTri) = false
+
+use_correction(::EpiPerTraceEntropyTri) = false
+
+function setup_extra_data(cone::EpiPerTraceEntropyTri{T}) where T
+    dim = cone.dim
+    cone.hess = Symmetric(zeros(T, dim, dim), :U)
+    cone.inv_hess = Symmetric(zeros(T, dim, dim), :U)
+    load_matrix(cone.hess_fact_cache, cone.hess)
+    cone.lwv = zeros(T, cone.d, cone.d)
+    cone.tau = zeros(T, cone.d, cone.d)
+    cone.W = zeros(T, cone.d, cone.d)
+    cone.w_vals_log = zeros(T, cone.d)
+    return cone
+end
+
+get_nu(cone::EpiPerTraceEntropyTri) = cone.d + 2
+
+function set_initial_point(arr::AbstractVector{T}, cone::EpiPerTraceEntropyTri{T}) where T
+    d = cone.d
+    (u, w) = get_central_ray_epiperentropy(d)
+    arr[1] = u
+    arr[2] = sqrt(T(d) * w * u + one(T))
+    for i in 1:d
+        arr[2 + sum(1:i)] = w
+    end
+    return arr
+end
+
+function update_feas(cone::EpiPerTraceEntropyTri{T}) where T
+    @assert !cone.feas_updated
+    u = cone.point[1]
+    v = cone.point[2]
+    @views W = Hermitian(svec_to_smat!(cone.W, cone.point[3:cone.dim], cone.rt2), :U)
+    (w_vals, w_vecs) = cone.fact_W = eigen(W)
+
+    if (v > eps(T)) && all(wi -> wi > eps(T), w_vals)
+        @. cone.w_vals_log = log(w_vals)
+        cone.lwv = w_vecs * Diagonal(cone.w_vals_log) * w_vecs'
+        lv = log(v)
+        for i in 1:cone.d
+            cone.lwv[i, i] -= lv
+        end
+        cone.z = u - dot(W, Symmetric(cone.lwv))
+        cone.is_feas = (cone.z > eps(T))
+    else
+        cone.is_feas = false
+    end
+
+    cone.feas_updated = true
+    return cone.is_feas
+end
+
+function is_dual_feas(cone::EpiPerTraceEntropyTri{T}) where T
+    u = cone.dual_point[1]
+    v = cone.dual_point[2]
+    @views W = Hermitian(svec_to_smat!(similar(cone.point, cone.d, cone.d), cone.dual_point[3:cone.dim], cone.rt2), :U)
+    if u > eps(T)
+        return v - u * tr(exp.(-W ./ u - I)) > eps(T)
```

Comment:
i think there's a mistake here

Reply:
maybe `tr(exp(-W / u - I))` (remove the dots)

Reply:
This should be simplified. But there are weird things happening in BigFloat.

Reply:
`tr(exp.(-W ./ u - I)) = sum(exp(-W[i,i] / u - 1) for i in 1:size(W,1))`
it ignores all but the diagonal elements of W

Reply:
oops yes the dot was a mistake. anyway i'll push a corrected function soon.

Reply:
there were other things wrong